src/lib.rs

```diff
@@ -0,0 +1,203 @@
+pub mod opts;
+
+use common_failures::Result;
+use crev_recursive_digest;
+use digest::Digest;
+use failure::bail;
+use failure::{format_err, ResultExt};
+use glob;
+use log::{debug, info, trace};
+use std::collections::HashSet;
+use std::path::Path;
+use std::path::PathBuf;
+
+fn get_coalesced_lines_from_dockerfile_content(content: String) -> Result<Vec<String>> {
+    let mut res = vec![];
+    let mut prev_line = String::from("");
+
+    for line in content
+        .lines()
+        .map(|line| line.trim())
+        .filter(|line| !line.starts_with("#"))
+    {
+        if line.ends_with("\\") {
+            prev_line.push_str(&line[..line.len() - 1]);
+        } else {
+            prev_line.push_str(line);
+            res.push(std::mem::replace(&mut prev_line, "".into()))
+        }
+    }
+
+    if prev_line != "" {
+        bail!("Trailing \\");
+    }
+
+    Ok(res
+        .into_iter()
+        .map(|line| line.trim().to_string())
+        .filter(|line| !line.is_empty())
+        .collect())
+}
+
+/// TODO: Bug, does not support `[src1, src2,... dst]` syntax
+fn get_external_paths_from_dockerfile_line(line: String) -> Result<Vec<PathBuf>> {
+    trace!("Long Line: {}", line);
+    let mut res = vec![];
+
+    let mut parts = line.split_ascii_whitespace();
+    let cmd = parts.next().expect("At least the command");
+
+    match cmd {
+        "COPY" | "ADD" => {
+            let src_globs: Vec<_> = parts.map(str::to_string).collect();
+            if src_globs.is_empty() {
+                bail!("No arguments to {} command?", cmd);
+            }
```

Comment:
Here you can use `ensure!` from the `failure` crate (which should be replaced IMO, but not in this PR. `failure` is deprecated, unmaintained, and was only a proof of concept). 
```suggestion
            failure::ensure!(!src_globs.is_empty(), "No arguments to {cmd} commands?");
```

Reply:
I don't know the Rust version being used to compile this, but hopefully it supports string interpolation.